### PR TITLE
More Fixes 

### PR DESCRIPTION
**Pull request by SteffenHeu**
Fixed some issues a colleague found when testing a bit:

- Feature table -> delete row did not delete the row (actually the feature table was just not updated after a row was deleted)
 - added a listener to update the table on changes to the list of rows
- Condition for show XIC (dialog) were incorrect
- No option to search in comments
 - added a combo box with every types of a feature list to filter for. It filters every type on a string basis like in MZmine 2. DataType.getFormattedString is used, so it filters as it's displayed in the table and therefore shouldnt lead to confusion with decimal digits not displayed in the table.
grafik

### CODE REVIEW

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFX.java [hunk 1]
```diff
@@ -565,4 +575,53 @@ public ModularFeature getSelectedFeature() {
...
+  /**
+   * Initialises a listener to update the tables' contents to the current feature list. Also adds
+   * and removes the row changed listener.
+   */
+  private void initFeatureListListener() {
+    featureListProperty().addListener((observable, oldValue, newValue) -> {
+      // Clear old rows and old columns
+      getRoot().getChildren().clear();
+      getColumns().clear();
+      rowItems.clear();
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFX.java:604]: put the whole thing in MZmineCore.runLater

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFX.java [hunk 2]
```diff
@@ -177,34 +179,34 @@ private void editFocusedCell() {
...
-    for (FeatureListRow row : flist.getRows()) {
-      ModularFeatureListRow mrow = (ModularFeatureListRow) row;
-//      logger.info("Add row with id: " + row.getID());
-      root.getChildren().add(new TreeItem<>(mrow));
-    }
-
-    rowItems.addAll(root.getChildren());
+  @Nonnull
+  private ListChangeListener<FeatureListRow> getRowListChangeListener(
+      TreeItem<ModularFeatureListRow> root) {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFX.java:189]: moved to class signature

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java [hunk 3]
```diff
@@ -88,6 +96,34 @@ public void initialize() {
 
     filtersRow.getChildren().addAll(filterIcon, mzFilter, separator, rtFilter);
 
+    typeComboBox = new ComboBox<>();
+    typeComboBox.setConverter(new StringConverter<>() {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:100]: is this the easiest way to do this?
Maybe we should just make DataType.toString return the headerString

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java [hunk 4]
```diff
@@ -88,6 +96,34 @@ public void initialize() {
...
+
+      @Override
+      public DataType fromString(String string) {
+        return null;
+      }
+    });
+    typeComboBox.valueProperty().addListener((observable, oldValue, newValue) -> filterRows());
+    featureTable.featureListProperty().addListener(((observable, oldValue, newValue) -> {
+      typeComboBox.setItems(
+          FXCollections.observableArrayList(newValue.getRowTypes().values()));
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:117]: i think there is no need in wrapping the values into an observableList. The values should already be observable: 
https://stackoverflow.com/questions/25842780/javafx-obsevablemap-keyset-as-an-observableset

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java [hunk 5]
```diff
@@ -116,28 +166,28 @@ private void filterRows() {
...
   private Range<Double> parseNumericFilter(TextField textField, double epsilon) {
     textField.setStyle("-fx-control-inner-background: #ffffff;");
     String filterStr = textField.getText();
-    filterStr = filterStr.replace(" ","");
+    filterStr = filterStr.replace(" ", "");
 
     if (filterStr.isEmpty()) { // Empty filter
       textField.setStyle("-fx-prompt-text-fill: derive(-fx-control-inner-background, -30%);");
       return RangeUtils.DOUBLE_INFINITE_RANGE;
     } else if (filterStr.contains("-")) { // Filter by range
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:186]: a negative value (-1) would trigger this. 
Maybe filterStr.indexOf("-")>0 to find a range idicator (-)

File: src/main/java/io/github/mzmine/util/javafx/TreeViewUtils.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+   * @param value The value.
+   * @param treeItems  The tree items to search.
+   * @param <T> The type.
+   * @return The tree item with the given value or null.
+   */
+  @Nullable
+  public static <T> TreeItem<T> getTreeItemByValue(T value,
+      List<TreeItem<T>> treeItems) {
+    for (var item : treeItems) {
+      if (item.getValue() == value) {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/util/javafx/TreeViewUtils.java:42]: at this point you should definitely use equals.
Maybe check if value is null and then value.equals

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java (hunk 3 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:100]: I'd say it's the best practice way.

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java (hunk 4 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:117]: But setItems requires a list.

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java (hunk 5 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:186]: Done.

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java (hunk 4 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:117]: ok

File: src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/featurelisttable_modular/FeatureTableFXMLTabAnchorPaneController.java:100]: What's to say against DataType.toString returning the header String?

**Comment by robinschmid** [comment]
All done. Thanks steffen

**Pull request merged**